### PR TITLE
Modify chart generator to support Kibana 7.x dashboards

### PR DESCRIPTION
**Pull request by ebadyano**
This commit changes the chart generator output to ndjson and addresses validation errors for child "query" on kibana 7.x.

### CODE REVIEW

**Comment by ebadyano** [comment]
This should only be merged once we upgrade our metric store to. 7.x.

**Comment by ebadyano** [comment]
@danielmitterdorfer @dliappis could you please review? Thank you!

**Comment by dliappis** [comment]
> @danielmitterdorfer @dliappis could you please review? Thank you!

could you fix the failing PR checks first? Thanks!

**Comment by ebadyano** [comment]
Sorry, fixed the lint issues.

File: esrally/chart_generator.py [hunk 1]
```diff
@@ -1587,7 +1587,7 @@ def bulk_tasks(self):
         task_names = []
         for task in self.track.find_challenge_or_default(self.challenge).schedule:
             for sub_task in task:
-                if sub_task.operation.type == track.OperationType.Bulk.to_hyphenated_string():
+                if track.OperationType.Bulk.to_hyphenated_string() in sub_task.operation.type:
```
Comment (danielmitterdorfer) [inline esrally/chart_generator.py:1590]: I think this warrants at least a comment and maybe even an info message to the terminal on lenient matches.

File: esrally/chart_generator.py [hunk 2]
```diff
@@ -1033,7 +1033,7 @@ def segment_memory(title, environment, race_config):
                         "seperate_axis": 1,
                         "split_mode": "filters",
                         "stacked": "none",
-                        "filter": "environment:{} AND track:{}".format(environment, race_config.track),
+                        "filter": "environment:{} AND track:\"{}\"".format(environment, race_config.track),
```
Comment (danielmitterdorfer) [inline esrally/chart_generator.py:1036]: Maybe we should change the `.format()` invocations with f-strings in all places?

**Review by danielmitterdorfer — COMMENTED** [review]
Thanks for the PR. I left a few comments / suggestions.

**Comment by danielmitterdorfer** [comment]
@elasticmachine test this please

File: esrally/chart_generator.py [hunk 3]
```diff
@@ -1484,7 +1471,7 @@ def generate_dashboard(chart_type, environment, track, charts, flavor=None):
                 "y": (row * height),
                 "w": chart_width,
                 "h": height,
-                "i": "{}".format(panelIndex)
+                "i": f"{panelIndex}"
```
Comment (danielmitterdorfer) [inline esrally/chart_generator.py:1474]: A `str(panelIndex)` should be sufficient here.

**Review by danielmitterdorfer — APPROVED** [review]
Thanks for iterating. I found one spot where we could simplify formatting a little but but apart from that LGTM. No need for another review round after addressing this.

**Comment by ebadyano** [comment]
@danielmitterdorfer thank you for the review!

**Pull request merged**